app/src/main/java/ch/epfl/sdp/appart/login/FirebaseLoginService.java

```diff
@@ -69,39 +70,45 @@ public User getCurrentUser() {
 
     @Override
     public CompletableFuture<User> createUser(String email, String password) {
+        if (getCurrentUser() != null) throw new IllegalStateException("the current user cannot be set");
         return handleEmailAndPasswordMethod(email, password,
                 mAuth.createUserWithEmailAndPassword(email, password));
     }
 
     @Override
     public CompletableFuture<Void> updateEmailAddress(String email) {
         if (email == null) throw new IllegalArgumentException();
+        if (getCurrentUser() == null) throw new IllegalStateException("current user must be set when updating the email");
         return setUpFuture(getCurrentFirebaseUser().updateEmail(email)
                 , result -> result);
     }
 
     @Override
     public CompletableFuture<Void> updatePassword(String password) {
         if (password == null) throw new IllegalArgumentException();
+        if (getCurrentUser() == null) throw new IllegalStateException("current user must be set when updating the password");
         return setUpFuture(getCurrentFirebaseUser().updatePassword(password),
                 result -> result);
     }
 
     @Override
     public CompletableFuture<Void> sendEmailVerification() {
+        if (getCurrentUser() == null) throw new IllegalStateException("current user must be set when sending verification mail");
         return setUpFuture(getCurrentFirebaseUser().sendEmailVerification(),
                 result -> result);
     }
 
     @Override
     public CompletableFuture<Void> deleteUser() {
+        if (getCurrentUser() == null) throw new IllegalStateException("current user must be set when deleting it");
         return setUpFuture(getCurrentFirebaseUser().delete(),
                 result -> result);
     }
 
     @Override
     public CompletableFuture<Void> reAuthenticateUser(String email, String password) {
         if (email == null || password == null) throw new IllegalArgumentException();
+        if (getCurrentUser() == null) throw new IllegalStateException("current user must be set when reAuthentication");

```

Comment:
En fait tous ces checks sont inutiles car les fonctions utilisent getCurrentFirebaseUser qui lance déjà une exception.

Reply:
yes but they add precision if the exception is thrown. Like the function that is being executed. Also I find it more readable.